addOns/reports/src/test/java/org/zaproxy/addon/reports/ExtensionReportsUnitTest.java

```diff
@@ -780,16 +785,42 @@ void checkAlert(JSONObject alert) {
         assertThat(alerts.getJSONObject(0).getString("sourceid"), is(equalTo("0")));
 
         JSONArray instances = alerts.getJSONObject(0).getJSONArray("instances");
-        assertThat(instances.size(), is(equalTo(1)));
+        assertThat(instances.size(), is(equalTo(2)));
+
+        checkJsonAlertInstance(instances, isJsonPlus, 0);
+        checkJsonAlertInstance(instances, isJsonPlus, 1);
+    }
+
+    private static void checkJsonAlertInstance(JSONArray instances, boolean isJsonPlus, int i) {
```

Comment:
Split the method (e.g. `checkJsonAlertInstance(…)` and `checkJsonAlertInstanceAndMessages(…)`) and call each as required (boolean should no longer be needed here).